.github/workflows/node.js.yml

```diff
@@ -53,24 +57,33 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-lintcache-
 
+      - name: Cache node modules
+        id: cache-node-modules
+        uses: actions/cache@v3
+        with:
+          path: |
+            node_modules
+            packages/*/node_modules
+          key: node-modules-${{ hashFiles('package-lock.json')}}
+
       - name: Install dependencies
+        if: steps.cache-node-modules.outputs.cache-hit != 'true'
         run: npm ci --no-audit
 
       - name: Build
-        run: npm run build
-
+        run: VITE_CORE_API_URL=http://localhost:10000/jsapi npm run build
       # Run unit tests on all files for accurate code coverage
       # Lint only changed files so linting is faster
       # Caching with the absolute path b/c Jest will make a folder in each project
       # Then there's caches in all package folders
-      - name: Test (Pull Request)
+      - name: Unit Test (Pull Request)
```

Comment:
Might be worth making unit tests a separate action that can run in parallel since it doesn't require build

Reply:
Good call, these can definitely run in parallel.

Reply:
With me splitting these into two different tests (e2e and unit), I'll need to update `infra` as well to make them both required status checks (instead of the one just called `build` now).

Reply:
I'll name one `build`, and update the description.